File: src/main/java/com/lessmarkup/interfaces/recordmodel/ModelCollection.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.interfaces.recordmodel;

import com.lessmarkup.interfaces.data.QueryBuilder;
import com.lessmarkup.interfaces.structure.NodeAccessType;
import java.sql.SQLException;
import java.util.Collection;
import java.util.List;
import java.util.OptionalLong;

public interface ModelCollection<T extends RecordModel> {
    List<Long> readIds(QueryBuilder query, boolean ignoreOrder);
    int getCollectionId();
    Collection<T> read(QueryBuilder queryBuilder, List<Long> ids);
    default void initialize(OptionalLong objectId, NodeAccessType nodeAccessType) {
    }
}

File: src/main/scala/com/lessmarkup/interfaces/recordmodel/ModelCollection.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.recordmodel

import com.lessmarkup.interfaces.data.QueryBuilder

trait ModelCollection[T] {
  def readIds(query: QueryBuilder, ignoreOrder: Boolean): Seq[Long]
  def getCollectionId: Int
  def read(queryBuilder: QueryBuilder, ids: Seq[Long]): Seq[T]
}
